### PR TITLE
Implemented authentication via EC2 instance role for ECR

### PR DESCRIPTION
**Pull request by sithmein**
This PR adds support for EC2 instance roles as described in issue #1177.
I didn't add any tests since they would only work when run on a correctly configured EC2 instance.

### CODE REVIEW

File: src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java [hunk 1]
```diff
@@ -217,13 +229,88 @@ private AuthConfig createStandardAuthConfig(boolean isPush, Map authConfigMap, S
...
+        		if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+        			// no instance role found
+					log.debug("No instance role found, return code was %d",
+							response.getStatusLine().getStatusCode());
+        			return null;
+        		}
+        		
+        		// read instance role
+                try (InputStream is = response.getEntity().getContent()) {
+                	instanceRole = IOUtils.toString(is, StandardCharsets.UTF_8);
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java]: any validation of this instance role needed ?

File: src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java [hunk 2]
```diff
@@ -217,13 +229,88 @@ private AuthConfig createStandardAuthConfig(boolean isPush, Map authConfigMap, S
...
+                }
+        	}
+        	log.debug("Found instance role %s, getting temporary security credentials", instanceRole);
+        	
+        	// get temporary credentials
+			request = new HttpGet(
+					"http://169.254.169.254/latest/meta-data/iam/security-credentials/" + instanceRole);
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java]: URL encoding required as instanceRole is an externally (aka 'tainted') data ?

File: src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java (hunk 1 above)
Comment (sithmein) [reply src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java]: What exactly do you mean with "validation" in this context?

File: src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java [hunk 3]
```diff
@@ -217,13 +230,88 @@ private AuthConfig createStandardAuthConfig(boolean isPush, Map authConfigMap, S
...
+					log.debug("No instance role found, return code was %d",
+							response.getStatusLine().getStatusCode());
+        			return null;
+        		}
+        		
+        		// read instance role
+                try (InputStream is = response.getEntity().getContent()) {
+                	instanceRole = IOUtils.toString(is, StandardCharsets.UTF_8);
+                }
+        	}
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java]: formatting seems to be messed up a bit

**Pull request merged**